whylog/tests/tests_log_reader/__init__.py

```diff
@@ -30,48 +31,73 @@ def _deduce_line_offset(self, file_path, line_no):
+                    LineSource("localhost", "node_1.log")
+                ) for line_str in eval(lines_str)
+            ]
+            results.append(InvestigationResult(lines, [{'name': constraint_str}]))
+            # there will not full info about constraint
+        return results
+
     @generate(
         '001_most_basic',
-        '002_match_latest',
+        # '002_match_latest',
         '003_match_time_range',
-        '005_match_tree',
+        # '005_match_tree',
         '006_match_parameter',
-        '007_match_or',
+        # '007_match_or',
         '008_match_and',
-        '009_match_negation',
-        '010_multiple_files',
-        '011_different_entry',
-        '012_multiple_rulebooks',
-        '013_match_and_incomplete',
+        # '009_match_negation',
+        # '010_multiple_files',
+        # '011_different_entry',
+        # '012_multiple_rulebooks',
+        # '013_match_and_incomplete',
     )
     def test_one(self, test_name):
+        # paths files setup
         prefix_path = os.path.join(*TestPaths.path_test_files)
         path = os.path.join(prefix_path, test_name)
         input_path = os.path.join(path, 'input.txt')
-        output_path = os.path.join(path, 'expected_output.txt')
+        # output_path = os.path.join(path, 'expected_output.txt')  # FIXME is it really unnecessary?
         log_file = os.path.join(path, 'node_1.log')
+        results_file = os.path.join(path, 'investigation_results.txt')
 
-        # TODO this 'if' is temporary, remove this later
-        if not test_name == '003_match_time_range':
-            raise SkipTest("Functionality not implemented yet")
-
+        # gathering information about effect line
         line_number = self._get_cause_line_number(input_path)
         line_content = self._get_concrete_line_from_file(log_file, line_number)
         effect_line_offset = self._deduce_line_offset(log_file, line_number)
 
+        # preparing Whylog structures, normally prepared by Front
         whylog_config = YamlConfig(*ConfigPathFactory.get_path_to_config_files(path))
         log_reader = LogReader(whylog_config)
         line = FrontInput(effect_line_offset, line_content, prefix_path)
 
-        result = log_reader.get_causes(line)
-
-        if test_name == '003_match_time_range':
-            # TODO this 'if' is temporary, remove this later
-            assert result
-            assert result[0].lines[0].line_content == self._get_last_line_from_file(output_path)
-            # ^ hack basing on that for test 003 last line of expected_output.txt is a cause line
-            # FIXME it is not appropriate for each test
-            # FIXME checking up correctness should be more advanced for more advanced tests
+        # action and checking the result
+        results = log_reader.get_causes(line)
+        assert results
+        true_results = self._investigation_results_from_file(log_file, results_file)
+        for got, real in zip(results, true_results):
```

Comment:
Before this line you should check that result and true_results have a same length